src/Utils/PromptNormalizer.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Utils;
+
+use WordPress\AiClient\Messages\DTO\Message;
+use WordPress\AiClient\Messages\DTO\MessagePart;
+use WordPress\AiClient\Messages\DTO\UserMessage;
+
+/**
+ * Utility class for normalizing various prompt formats into standardized Message arrays.
+ *
+ * @since n.e.x.t
+ */
+class PromptNormalizer
```

Comment:
I have a very similar idea implemented in https://github.com/WordPress/php-ai-client/pull/39/files#diff-6593318e4cf91319bc205a2a05d02dfee870c630eba8ddccb249c9c51243a5d7R19. The names are different, but it's the same idea.

I believe my implementation is more flexible than the one here, but I like your naming. Maybe you can take a look at mine and incorporate / reuse logic in here where it makes sense?

Reply:
Resolved in bdb86eb